### ISSUE TITLE
Query regarding INI simulator

### ISSUE THREAD

**Issue by sauravtii** [NeuromorphicProcessorProject/snn_toolbox#128]
I tried simulating my convoltional spiking neural network on INI using snntoolbox and it was successful. I just had a couple of questions, so can anyone please answer them ?

1. Is there any underlying neuromorphic hardware for INI simulator that it uses to carry out the simulation or it just does it using CPU ?
2. Can I use INI without using SNNtoolbox or in order to use INI I will have to use SNNtoolbox ?
3. Are there any resources that would tell me the differences between INI and other simulators for spiking neural networks ?

**Comment by rbodo** [comment]
1. GPU / CPU
2. An SNN converted for the INI backend is just a tensorflow DNN with added functionality (such as membrane potential, spike behavior etc). So in principle you can run the resulting graph outside of the toolbox but then you have to take care of the right input encoding and writing your own simulation time loop.
3. https://snntoolbox.readthedocs.io/en/latest/guide/intro.html

**Comment by sauravtii** [comment]
Thanks for your response @rbodo. Is it possible to get the power metrics for INI using snntoolbox ?

**Comment by rbodo** [comment]
The toolbox has no power measurement built in for INI. Since it's just a DNN running on GPU, you can use standard benchmarking tools for that.

The toolbox supports pynn but provides link to a BrainScaleS backend. One option would be to run the conversion with nest / Brian2 backend and then use the saved weight files when constructing your model on BrainScaleS directly. These files are just generic connection tables. The toolbox handles the unrolling of the convolution layers into fully-connected layers, and optionally performs weight normalization to improve the SNN accuracy. You should be able to use the resulting connection tables in a network of fully-connected non-leaky integrate-and-fire neurons. You might notice a discrepancy on neuromorphic hardware though because of weight / state quantization.

**Comment by sauravtii** [comment]
Okay. Is there any converter available in SNN toolbox that justs converts the SNN in pynn format and when I am using BrainScaleS using EBRAINS, I can just build up the model using saved configuration (saved model in the pynn format) and therfore run my SNN on BrainScaleS ?
Also, if you know the code or any resource that achieves this thing, can you please share it ?

**Comment by rbodo** [comment]
> Do you mean SNN in "just a DNN running on GPU" ?

Yes, in case of INI sim (see my first answer).

I don't know of a resource that does conversion for Brainscales, and don't know how to "build up the model" in their API. Perhaps best to contact their team.

**Comment by sauravtii** [comment]
Okay, thank you so much for your help!
